app/controllers/shipping_api_controller.rb

```diff
@@ -0,0 +1,64 @@
+class ShippingApiController < ApplicationController
+  skip_before_filter :verify_authenticity_token, only: [:rates, :create_log]
+  before_action :set_shipping_rate, only: :rates
+
+  def rates
+    if @shipping_rate.valid?
+      estimates = @shipping_rate.ups_rates + @shipping_rate.usps_rates
+      render json: estimates, status: :ok # TODO: add status code tests
+    else
+      render json: @shipping_rate.errors, status: :bad_request # TODO: add status code tests
+
+    end
+  end
+
+  def create_log
```

Comment:
Does this method have anything to do with rates? Seems like maybe is should be a separate controller, like `LogsController#create`. Following the Rails convention of mapping models to controllers by name helps you find the method later when you need to change it or fix a bug.